### ISSUE TITLE
yield should not be allowed as an identifier inside a generator body

### ISSUE THREAD

**Issue by ariya** [eslint/espree#191]
While linting some (sloppy) code, I found out that there is no error on `yield` as demonstrated in the following reduced test case:

```js
espree.parse('function *g() { var yield; }', { ecmaFeatures: { generators: true } })
```

Actual: a syntax tree.

Expected: an error or an exception.

**Comment by nzakas** [comment]
Fixed in 3.0.0.
